lib/money/currency.rb

```diff
@@ -243,12 +264,11 @@ def stringify_keys
     # @example
     #   Money::Currency.new(:usd) #=> #<Money::Currency id: usd ...>
     def initialize(id)
-      id = id.to_s.downcase
-      unless self.class.stringified_keys.include?(id)
-        raise UnknownCurrency, "Unknown currency '#{id}'"
-      end
       @id = id.to_sym
       initialize_data!
+      self.class.instances_mutex.synchronize do
+        self.class.instances[id] = self
+      end
```

Comment:
This won't help. Cause `.new` still be able to run twice. I think it's not a big deal to have few Currency instances in this case, while it's threadsafe and there should not be errors. If we really want to have same instances, we should use `thread_safe` gem with smth like this
```ruby
alias_method :original_new, :new
def new(id)
  instances[id]
end

@instanses = ThreadSafe::Hash.new { |h, k| h[k] = original_new(k) }
# I'm not really sure if it checks and calls block atomically
```